toolkit/offchain/src/d_param/mod.rs

```diff
@@ -136,24 +131,38 @@ where
 		policy,
 		d_parameter,
 		&ctx,
+		zero_ex_units.clone(),
 		zero_ex_units,
 		gov_utxo.clone(),
 	)?;
+
 	let evaluate_response = client.evaluate_transaction(&tx.to_bytes()).await.map_err(|e| {
 		anyhow!(
 			"Evaluate insert D-parameter transaction request failed: {}, bytes: {}",
 			e,
 			hex::encode(tx.to_bytes())
 		)
 	})?;
-	let mut mint_witness_ex_units = get_validator_budgets(evaluate_response);
-	mint_witness_ex_units.mint_ex_units.reverse();
+
+	let ogmios_gov_script =
+		gov_utxo.clone().script.expect("Gov UTXO should have a reference script");
+	let gov_policy = PlutusScript::from_ogmios(ogmios_gov_script)?;
+	let mint_keys = tx.body().mint().expect("insert D parameter transaction has two mints").keys();
+	let script_to_index: HashMap<ScriptHash, usize> =
+		vec![(mint_keys.get(0), 0), (mint_keys.get(1), 1)].into_iter().collect();
+	let mint_ex_units = get_validator_budgets(evaluate_response).mint_ex_units;
+	let policy_idx = script_to_index.get(&policy.csl_script_hash()).unwrap().clone();
+	let gov_policy_idx = script_to_index.get(&gov_policy.csl_script_hash()).unwrap().clone();
+	let policy_ex_units = mint_ex_units.get(policy_idx).unwrap();
```

Comment:
I'm okay with the previous `unwraps` (however `expect` could be used), but this and the following I would for sure prefer to have an information that Ogmios answer (`mint_ex_units`) were supposed to have two items.
Alternatively, justify these `unwrap`s with a check on `evaluate_response`.